Tema-Java8/src/main/java/java8/Java8Main.java

```diff
@@ -0,0 +1,278 @@
+        Optional<Flower> optionalOfInstance = Optional.ofNullable(new Flower("Carnation", "pink", 40, 60, 60));
+        if (optionalOfInstance.isPresent()) {
+            System.out.println("Optional if isPresent checks out: " + optionalOfInstance);
+        }
+
+        //I emptied it to see what it prints
+        optionalOfInstance = Optional.empty();
+        System.out.println("The emptied one: " + optionalOfInstance);
+
+
+        //this throws a NullPointerException that's why we use ofNullable method below
+//        Optional<String> optionalNull = Optional.of(null);
+//        System.out.println(optionalNull);
+
+        // the method ofNullable returns an optional of specified values, otherwise, it returns an empty Optional
+        Optional<String> optionalOfNullable = Optional.ofNullable(null);
+        System.out.println("With ofNullable method: " + optionalOfNullable);
+    }
+
+    /**
+     * 9)
+     *
+     * @param flowers the list of filtered instances by numberOfPetals
+     * @return how many instances' field (numberOfPetals) "match" my condition (if it's an even number)
+     */
+    private static long countOfEvenNumberOfPetals(List<Flower> flowers) {
+        return flowers.stream().filter(x -> x.getNumberOfPetals() % 2 == 0).count();
+    }
+
+
+    /**
+     * 8)
+     * I chose to sort my list by height field then printed it
+     *
+     * @param flowers the list that I want to sort
+     * @return the sorted list
+     */
+    private static List<Flower> sortFlowersByHeight(List<Flower> flowers) {
+        return flowers.stream().sorted(Comparator.comparing(Flower::getHeight)).collect(Collectors.toList());
+
+    }
+
+    /**
+     * 7)
+     * I printed all the random strings from the set in reverse order using the reverseOrder() method.
+     * As you can see, the "max" string from the forth problem (maxOfRandomStrings method) is the first one printed.
+     * Also, I tested it myself with a set of easier to understand strings before implementing
+     *
+     * @param randomString the set of random strings generated below
+     */
+    private static void sortRandomStringsSetInReverseOrder(Set<String> randomString) {
+
+        System.out.println("The reverse sorted strings are: ");
+        randomString.stream()
+                .sorted(Comparator.reverseOrder())
+                .forEach(System.out::println);
+
+        System.out.println("\n");
+    }
+
+    /**
+     * 6)
+     * I used the model from the map below, then with a filter, counted the values greater than 10
+     *
+     * @return count the number of values of the map that are greater than 10
+     */
+    private static long howManyAreGreaterThanTen() {
+        Map<Integer, Integer> map = mapOfRandomIntegersAndTheirSquaredNumber(5);
+
+        System.out.println("The new map made for counting the values that are greater than 10: " + map);
+        return map.entrySet().stream().filter(mapp -> mapp.getValue() > 10).count();
```

Comment:
format: .stream() new line, .filter() new line...easier to read (code convention), mapp (?) => mapStream